### DISCUSSION TITLE
There is cluster (high availability) setting in 'keycloak by Quarkus' like 'legacy keycloak`

### DISCUSSION THREAD

**Discussion by WonJuneLee** [keycloak/keycloak#13600] [Q&A] [answered]
Hi all.

I found [documentation](https://www.keycloak.org/docs/latest/server_installation/index.html#_standalone-ha-mode)  for HA cluster setup.

Looks like I need to set the cluster mode, ping and architecture.
But that's a legacy keycloak document.

So I looked for an option or guide for Quarkus Keycloak. But the only thing is a guide to [caching](https://www.keycloak.org/server/caching).

- Is Distributed Cache configuration the only clustering option?
- Does Quarkus Keycloak run based on the ping of the connected DB, so no special setup is required?
- If I turn off the cache, (even though I lost some performance)I don't need to set up clustering and I only need to attach one RDBMS table?

Sorry for the basic question, thanks for the answer

**Comment by darius-m** [comment, marked as answer]
I think I can answer the last question - at least in the legacy (WildFly) configuration, user sessions are not kept in the database at all; if the Infinispan cluster fails you will get inconsistent service configurations and dropped user sessions (e,g., if the Infinispan servers do not have any cache persistence configured, all users will have to re-authenticate if all Infinispan servers are restarted at the same time). Because of this, you cannot create a Keycloak cluster that does not rely on Infinispan to store data.

The database mostly keeps information about more permanent service configurations (e.g., user 2FA tokens, client configurations), but not all information is placed in the database.

**Reply by darius-m** [reply]
Hopefully someone will correct me if there is a way to not use caches at all; most issues that we have faced were caused by Infinispan nodes disconnecting from the cluster and likely creating a split brain.
